tsconfig.json

```diff
@@ -1,12 +1,39 @@
 {
-  "compilerOptions": {
-    "target": "esnext",
-    "moduleResolution": "node",
-    "emitDeclarationOnly": true,
-    "strict": true,
-    "esModuleInterop": true,
-    "declaration": true,
-    "declarationDir": "build/esmodules"
-  },
-  "include": ["src"]
-}
+	"compilerOptions": {
+		"target": "esnext",
+		"moduleResolution": "node",
+		"emitDeclarationOnly": true,
+		"strict": true,
+		"esModuleInterop": true,
+		"declaration": true,
+		"declarationDir": "build/esmodules",
+		"useDefineForClassFields": true,
+		"lib": [
+			"esnext",
+			"ES2020",
+			"DOM",
+			"DOM.Iterable"
+		],
+		"module": "ESNext",
+		"skipLibCheck": true,
+		/* Bundler mode */
+		"allowImportingTsExtensions": true,
+		"resolveJsonModule": true,
+		"isolatedModules": true,
+		"jsx": "react",
+		/* Linting */
+		"noUnusedLocals": true,
+		"noUnusedParameters": true,
+		"noFallthroughCasesInSwitch": true,
+		"noImplicitOverride": true,
+		"noImplicitReturns": true,
+		"noPropertyAccessFromIndexSignature": true,
+		"noUncheckedIndexedAccess": true
+	},
+	"include": [
+		"src",
+		"e2e",
+		"public",
+		"tools",
+	]
```

Comment:
Some of these don't exist

Reply:
Removing e2e, other three should be included (I think?)